### ISSUE TITLE
Problems with the search engine

### ISSUE THREAD

**Issue by arno1979** [daneren2005/Subsonic#673]
As I posted on the [forum](http://forum.subsonic.org/forum/viewtopic.php?f=16&t=16528) I noticed two problems with the search function of DSub:

1. If the for instance an album title am searching for contains an apostrophe (') it will not be found.
In my case the album title is "I'm not there" (a film soundtrack containing tracks by various artists.

2. When searching for example for "Frank Zappa" the search works fine. If I search only for "Zappa" no results are found.
It looks like the value stored in the database has to start with the beginning of the search string.

Both these problems do not occur on the web interface of my ampache installation.
I don't know if this is specific to the subsonic backend of ampache or if the problem lies within DSub, but I suspect the latter that's why I am reporting the issue here.

Thanks for any ideas/help!

Arno

**Comment by daneren2005** [comment]
I haven't tested 1 yet but I can verify 2 is a problem with Ampache.  I search for made against Subsonic and "American Made" showed up but in Ampache it does not.  Ampache probably isn't routing the searches through the same methods for the REST API as it is for the web interface.

**Comment by mavenius** [comment]
I just tested against the latest beta subsonic server, and searching for
"life's" brought back results for "life on life's terms" so I don't think
this is a dsub issue.

Thank you,

Mark Avenius
Mark@Aveni.us

As I posted on the forum
<http://forum.subsonic.org/forum/viewtopic.php?f=16&t=16528> I noticed two
problems with the search function of DSub:

   1.

   If the for instance an album title am searching for contains an
   apostrophe (') it will not be found.
   In my case the album title is "I'm not there" (a film soundtrack
   containing tracks by various artists.
   2.

   When searching for example for "Frank Zappa" the search works fine. If I
   search only for "Zappa" no results are found.
   It looks like the value stored in the database has to start with the
   beginning of the search string.

Both these problems do not occur on the web interface of my ampache
installation.
I don't know if this is specific to the subsonic backend of ampache or if
the problem lies within DSub, but I suspect the latter.

Thanks for any ideas/help!

Arno

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub
<https://github.com/daneren2005/Subsonic/issues/673>

**Comment by arno1979** [comment]
I just found out on the ampache page's issue [https://github.com/ampache/ampache/issues/1198] that it's only the instant search that doesn't seem to work correctly. If I actually hit the search button the search for "Zappa" actually returns "Frank Zappa".
Just thought I'll let you know...

**Comment by arno1979** [comment]
https://github.com/ampache/ampache/issues/1198 has been closed so I was hoping this issue could be reopened because the problem still exists and at this point. Please have a look at my previous post for details.
Thanks!

**Comment by daneren2005** [comment]
I have fixed #1 with https://github.com/daneren2005/Subsonic/commit/9655f0cfdef513793c11560632b0652679412d6a.  Now I can't seem to reproduce #2 anymore.  I tried searching `American Made` again and now it returns the results.  I am not sure if I accidentally fixed it or not but I don't remember making any changes which would have affected that.  I will have to investigate more.  Or if you want to email me a username/password to dsub.android@gmail.com I can double check whether or not `Frank Zappa` works on your server now with my development version of DSub.
